autogtp/Production.cpp

```diff
@@ -41,29 +43,49 @@ void ProductionWorker::run() {
                  return;
              }
              game.readMove();
-         } while (game.nextMove());
-         QTextStream(stdout) << "Game has ended." << endl;
-         if (game.getScore()) {
-             game.writeSgf();
-             game.dumpTraining();
+         } while (game.nextMove() && m_state == RUNNING);
+         switch(m_state) {
+         case RUNNING:
+         {
+             QTextStream(stdout) << "Game has ended." << endl;
+             if (game.getScore()) {
```

Comment:
Here you only writeSgf if (getScore)
on line 64 you always write the file,

should they use the same logic?